agent/api/container/container.go

```diff
@@ -1270,3 +1270,27 @@ func (c *Container) UpdateManagedAgentSentStatus(agentName string, status apicon
 	}
 	return false
 }
+
+func (c *Container) GetManagedAgentStatus(agentName string) apicontainerstatus.ManagedAgentStatus {
+	c.lock.Lock()
```

Comment:
Let's change this to a read lock/unlock (`c.lock.RLock()`), same for `GetManagedAgentSentStatus`